exercises/binary-search/.meta/solutions/binary_search.rb

```diff
@@ -2,21 +2,22 @@ class BinarySearch
   attr_reader :list
 
   def initialize(data)
-    fail ArgumentError unless data.sort == data
     @list = data
   end
 
   def search_for(datum)
+    return nil if list.empty?
     return middle if list[middle] == datum
 
     if list[middle] > datum
-      sublist = list[0..middle]
-      fail 'Not Found' if sublist == list
+      sublist = list[0...middle]
+      return nil if sublist == list
       return BinarySearch.new(sublist).search_for(datum)
     else
       sublist = list[middle..-1]
-      fail 'Not Found' if sublist == list
-      return BinarySearch.new(sublist).search_for(datum) + middle
+      return nil if sublist == list
+      result = BinarySearch.new(sublist).search_for(datum)
+      return result.nil? ? nil : result + middle
```

Comment:
The example solution doesn't need to be idiomatic, or even clean.  Better ways to solve this will come out with students solving the new version... and if we see this solution, we may be able to realize the student has found it on github, and has submitted it, and so we can talk about the ways to improve this code with them at this time.  So I don't usually look too closely at the example solutions for this, as long as they do pass the tests.  The tests are more important to be exemplary Ruby, as that is delivered to the students.